### PR TITLE
ICP-9214 Fix fan event handling for GE controller

### PR DESCRIPTION
**Pull request by dkirker**
This is a first pass. Not fully ready for review yet.

### CODE REVIEW

**Comment by dkirker** [comment]
Waiting for feedback from QA about the other GE device and the Honeywell device.

**Comment by dkirker** [comment]
Updated given the testing/verification that I did with the Leviton and Honeywell models

File: devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy
Comment (tpmanley) [inline devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy]: What other values do they report? I just wonder if it would be better to change the code above to do `Math.ceil((cmd.value + 1)/33) as int`

Comment (greens) [reply devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy]: Have never been able to get a consistent answer about that, hence all the changes to this DTH

**Review by greens — DISMISSED** [review]

File: devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy
Comment (dkirker) [reply devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy]: 1-32 = 1,
33-66 = 2,
67 - 99 or 100 = 3

Bashi realized that calling setLevel could be used as an indicator as to what that value means to the bulb.

Comment (dkirker) [reply devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy]: I suspect the firmware does a check on the values and not some math.

if 1 <= i <= 32

else if 33 <= i <= 66

else

🤷‍♂

Comment (tpmanley) [reply devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy]: Personally I think the range checks would be preferable over the use of ceil and the special check for 33

Comment (dkirker) [reply devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy]: @tpmanley I am going to give this a whirl today. Made some updates.

**Review by tpmanley — APPROVED** [review]

**Review by greens — APPROVED** [review]

**Review by kianooshkarami — APPROVED** [review]

**Comment by dkirker** [comment]
@tpmanley I had pointed this at staging in anticipation to have it in prod by now. We should probably merge this tomorrow after the deploy. I am doing a test of the code today to verify my recent change.

**Comment by dkirker** [comment]
@tpmanley  Verified yesterday. Since the deploy happened today we can merge this so it is in staging. Could you please merge?

**Pull request merged**
